new documentation on type inferring with fq (#13807)
ydb-platform/ydb: ydb/docs/ru/core/concepts/federated_query/s3/external_data_source.md

Year Int32 NOT NULL
```

> Если поле данных помечено как обязательное, `NOT NULL`, но это поле отсутствует в обрабатываемом файле, то работа с таким файлом будет завершена с ошибкой. Если поле помечено как необязательное, `NULL`, то при отсутствии поля в обрабатываемом файле не будет возникать ошибки, но поле при этом примет значение `NULL`. Ключевое слово `NULL` в необязательных полях опционально.

Reviewer: В duckdb например подробно описывают как типы выводятся, какие приоритеты, форматы у типов должны быть. Мы не хотим подробнее описать как он будет выводить типы? https://duckdb.org/docs/data/csv/auto_detection

> Если поле данных помечено как обязательное, `NOT NULL`, но это поле отсутствует в обрабатываемом файле, то работа с таким файлом будет завершена с ошибкой. Если поле помечено как необязательное, `NULL`, то при отсутствии поля в обрабатываемом файле не будет возникать ошибки, но поле при этом примет значение `NULL`. Ключевое слово `NULL` в необязательных полях опционально.

Reviewer 2: ```suggestion
Если поле данных помечено как обязательное, `NOT NULL`, но это поле отсутствует в обрабатываемом файле, то работа с таким файлом будет завершена с ошибкой. Если поле помечено как необязательное, `NULL`, то при отсутствии поля в обрабатываемом файле ошибки не возникнет, но поле примет значение `NULL`. Ключевое слово `NULL` в необязательных полях является опциональным.
```

> ### Автоматический вывод схемы данных {#inferring}

Reviewer 2: ```suggestion
### Автоматический вывод схемы данных {#inference}
```

> Автоматический вывод схемы доступен для всех [форматов данных](formats.md#formats), кроме `raw` и `json_as_string`. Он удобен в случаях, когда схема содержит большое количество полей. Чтобы не вводить эти поля вручную, используйте параметр `WITH_INFER`:

Reviewer 3: {{ ydb-short-name }} может автоматически определять схему данных в файлах бакета, чтобы вам не пришлось указывать все поля схемы вручную.

{% note info %}

Автоматический вывод схемы доступен для всех [форматов данных](formats.md#formats), кроме `raw` и `json_as_string`. Для этих форматов придётся [описывать схему данных вручную](#schema). 

{% endnote %}

Чтобы включить автоматический вывод схемы, используйте параметр `WITH_INFER`:

> Автоматический вывод схемы доступен для всех [форматов данных](formats.md#formats), кроме `raw` и `json_as_string`. Он удобен в том случае, когда схема содержит большое количество полей. Для того, чтобы не вводить эти поля вручную, используйте параметр `WITH_INFER`:

Reviewer 2: ```suggestion
Автоматический вывод схемы доступен для всех [форматов данных](formats.md#formats), кроме `raw` и `json_as_string`. Он удобен в случаях, когда схема содержит большое количество полей. Чтобы не вводить эти поля вручную, используйте параметр `WITH_INFER`:
```

```yql
SELECT
  <expression>
FROM
  <object_storage_connection_name>.`<file_path>`
WITH(
  FORMAT = "<file_format>",
  COMPRESSION = "<compression>",
  WITH_INFER = "true")
WHERE
  <filter>;
```

Где:

* `object_storage_connection_name` — название внешнего источника данных, ведущего на бакет с S3 ({{ objstorage-full-name }}).
* `file_path` — путь к файлу или файлам внутри бакета. Поддерживаются wildcards `*`, подробнее [в разделе](#path_format).

> * `file_format` — [формат данных](formats.md#formats) в файлах.

Reviewer: Может имеет смысл еще тут добавить информацию что `raw` и `json_as_string` не поддерживается. А почему мы raw кстати не поддержали, можешь напомнить?

> Где:
> 
> * `object_storage_connection_name` — название внешнего источника данных, ведущего на бакет с S3 ({{ objstorage-full-name }}).
> * `file_path` — путь к файлу или файлам внутри бакета. Поддерживаются wildcards `*`, подробнее [в разделе](#path_format).
> * `file_format` — [формат данных](formats.md#formats) в файлах. Поддерживаются все форматы, кроме `raw` и `json_as_string`.
> * `compression` — [формат сжатия](formats.md#compression_formats) файлов.
> 
> В результате выполнения такого запроса будут автоматически выведены названия и типы полей.

Reviewer 2: ```suggestion
Где:

* `object_storage_connection_name` — название внешнего источника данных, ведущего на S3 бакет ({{ objstorage-full-name }}).
* `file_path` — путь к файлу или файлам внутри бакета. Поддерживаются wildcards `*`, подробнее [ниже](#path_format).
* `file_format` — [формат данных](formats.md#formats) в файлах. Поддерживаются все форматы, кроме `raw` и `json_as_string`.
* `compression` — [формат сжатия](formats.md#compression_formats) файлов.

В результате выполнения такого запроса будут автоматически выведены названия и типы полей.
```
